common/src/main/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoader.java

```diff
@@ -259,25 +259,25 @@ private void saveSecretKeyToStorage(@NonNull final SecretKey unencryptedKey) thr
         KeyPair keyPair = AndroidKeyStoreUtil.readKey(mAlias);
         if (keyPair == null) {
             Logger.info(methodTag, "No existing keypair. Generating a new one.");
-            final Span span = OTelUtility.createSpan(SpanName.KeyPairGeneration.name());
-            final long keypairGenStartTime = System.currentTimeMillis();
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && CommonFlightsManager.INSTANCE.getFlightsProvider().isFlightEnabled(CommonFlight.ENABLE_NEW_KEY_GEN_SPEC_FOR_WRAP)) {
+                final long keypairGenStartTime = System.currentTimeMillis();
+                final Span span = OTelUtility.createSpanFromParent(SpanName.KeyPairGeneration.name(), SpanExtension.current().getSpanContext());
                 try (final Scope scope = SpanExtension.makeCurrentSpan(span)) {
                     keyPair = attemptKeyPairGeneration(mAlias, true, keypairGenStartTime);
                     Logger.info(methodTag, "Successfully generated keypair with new KeyPairGeneratorSpec with wrap purpose.");
-                    Span.current().setAttribute(AttributeName.key_pair_gen_successful_method.name(), "new_key_gen_spec_with_wrap");
+                    span.setAttribute("key_pair_gen_successful_method", "new_key_gen_spec_with_wrap");
```

Comment:
Why hard coding attribute name here? Why not declare it in AttributeName.java file?

Reply:
Oh happened by mistake. changing it.